### PR TITLE
Enable locale picker in web

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
This PR provides a Browser implementation of the `Configure Display Language` command which uses `localStorage` to store the locale:

https://user-images.githubusercontent.com/2644648/170383874-130859dc-2663-4eab-a2d0-5326d0525d5a.mp4

#### Follow up

* Update vscode.dev to honor this localStorage update

### CODE REVIEW

File: src/vs/base/common/platform.ts [hunk 1]
```diff
@@ -80,7 +85,7 @@ if (typeof navigator === 'object' && !isElectronRenderer) {
 	_isIOS = (_userAgent.indexOf('Macintosh') >= 0 || _userAgent.indexOf('iPad') >= 0 || _userAgent.indexOf('iPhone') >= 0) && !!navigator.maxTouchPoints && navigator.maxTouchPoints > 0;
 	_isLinux = _userAgent.indexOf('Linux') >= 0;
 	_isWeb = true;
-	_locale = navigator.language;
+	_locale = window.localStorage.getItem('vscode.nls.locale') || navigator.language;
```
Comment (TylerLeonhardt) [inline src/vs/base/common/platform.ts:88]: For some reason, this addition in this file causes [this integration test](https://github.com/microsoft/vscode/blob/b297fc96b45c26d9cf77e08421ef4f9f0ca99cf2/extensions/vscode-api-tests/src/singlefolder-tests/languages.test.ts#L99) to fail constantly and I'm not sure why...

**Comment by TylerLeonhardt** [comment]
I think I forgot a change in workbench.html that uses the `vscode.nls.locale` value instead of `navigator.language` if it's set. Just imagine that that change is included in this PR. I'll add it later today.

File: src/vs/base/common/platform.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/vs/base/common/platform.ts:88]: Perhaps I can get this value another way.. like through the AMD loader's configuration.

File: src/vs/base/common/platform.ts [hunk 2]
```diff
@@ -80,7 +80,19 @@ if (typeof navigator === 'object' && !isElectronRenderer) {
...
+	// Gather loader configuration since that contains the locale
+	let loaderConfiguration: any = null;
+	if (typeof globals.require !== 'undefined' && typeof globals.require.getConfig === 'function') {
+		// Get the configuration from the Monaco AMD Loader
+		loaderConfiguration = globals.require.getConfig();
+	} else if (typeof globals.requirejs !== 'undefined') {
+		// Get the configuration from requirejs
+		loaderConfiguration = globals.requirejs.s.contexts._.config;
+	}
+	const configuredLocale = loaderConfiguration?.['vs/nls']?.['availableLanguages']?.['*'] as string | undefined;
```
Comment (TylerLeonhardt) [inline src/vs/base/common/platform.ts:93]: @alexdima do you know if this is this the best way to get the locale out of the loader configuration?

Comment (alexdima) [reply src/vs/base/common/platform.ts:93]: You could export proper API for it via `vs/nls`. Today, `vs/nls` exports a function called `localize`, as well as other functions like `.setPseudoTranslation`, `.create`, `.load`. You could export via proper API from `vs/nls` the current configured locale. If you decide to do it, then please also add a mock implementation for it at `nls.mock.ts`.

**Review by alexdima — APPROVED** [review]

File: src/vs/base/common/platform.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/vs/base/common/platform.ts:93]: Ooo this is a great idea!

**Comment by TylerLeonhardt** [comment]
merging this in now just so that it's easier for me to make other changes

**Pull request merged**